src/dvclive/lightning.py

```diff
@@ -81,12 +81,16 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None):
         ), "experiment tried to log from global_rank != 0"
 
         self.experiment.step = step
+        epoch = False
         for metric_name, metric_val in metrics.items():
             if is_tensor(metric_val):
                 metric_val = metric_val.cpu().detach().item()
+            if metric_name.endswith("_epoch"):
+                epoch = True
```

Comment:
Take this with a grain of salt (because the more I look into [logger_connector](https://github.com/Lightning-AI/lightning/blob/master/src/lightning/pytorch/trainer/connectors/logger_connector/logger_connector.py), the more confused I am 😅 ), but I think this should be:

```suggestion
            if metric_name == "epoch":
```

It appears that the inclusion of `_step`/`_epoch` suffix depends on how people set up their `log` calls, so we might fail to set `epoch=True`.

However, lightning logs the `epoch` value as a metric, and it seems to align with the frequency specified with `log_every_n_steps`.

Reply:
Ok I think what we want is to only calle next_step if `self.experiment.step != step` (this need to be checked at the top)

Reply:
I think instead what we want is to exclude anything that ends with `_step`. WDYT?

Reply:
I mentioned the `step` logic because another fun thing is that other things can call `log_metrics` separately but with the same step value example, the [LearningRateMonitor](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.LearningRateMonitor.html)

Reply:
> `_step`/`_epoch` prefixes are only included if both `on_step=True, on_epoch=True` has been passed to `self.log`.

Yeah, but I think that's fine because it's only a problem in that scenario. If I have `on_step=False`, the logging is already reasonable. If I have `on_epoch=False`, I think we have no choice but to log everything (I doubt it's very common). If we only skip on `metric_name.endswith("_step")`, it's pretty safe while still reducing the noise for `on_step=True, on_epoch=True` scenarios.